src/hpr_routing.erl

```diff
@@ -92,25 +86,31 @@ deliver_packet(Packet, [Route | Routes]) ->
         "delivering packet to ~s",
         [hpr_route:lns(Route)]
     ),
-    %% FIXME: delivery could be halted to multiple routes if one of the earlier
-    %% Protocol:send(...) errors out.
-    Resp =
-        case Protocol of
-            {packet_router, _} ->
-                hpr_protocol_router:send(Packet, self(), Route);
-            {gwmp, _} ->
-                hpr_protocol_gwmp:send(Packet, self(), Route);
-            _OtherProtocol ->
-                lager:warning([{protocol, _OtherProtocol}], "unimplemented"),
-                ok
-        end,
-    case Resp of
+    Key = crypto:hash(sha256, <<
+        (hpr_packet_up:phash(Packet))/binary, (hpr_route:lns(Route))/binary
+    >>),
```

Comment:
This answer my previous comment about allowing different numbers of packet per OUI.

I don't think it would be a widespread problem, but putting here for posterity sake because I'm thinking about it.

If for some reason, someone made multiple routes that went to the same endpoint.

e.g.
- Wildcard over the same JoinEUI, but split DevAddrs for easier activation of routing. For those joins, they would get half the amount they expected, but twice.

- If someone had a set of routes that happened to overlap DevAddr addresses (whether on purpose, or not) those would also get half the packets they were expecting, but twice.

Reply:
I am thinking that we should probably add some kind of unique ID per route. It would save this problem as well as allow us to do individual updates.